app/models/mdm/payload.rb

```diff
@@ -0,0 +1,98 @@
+class Mdm::Payload < ActiveRecord::Base
+  extend ActiveSupport::Autoload
+
+  include Metasploit::Model::Search
+
+  #
+  # Associations
+  #
+
+  # {Mdm::Workspace} in which this payload was created.
+  belongs_to :workspace,
+             class_name: 'Mdm::Workspace',
+             inverse_of: :payloads
+
+
+  #
+  # Attributes
+  #
+
+  # @!attribute [rw] name
+  #   The name of this payload.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] uuid
+  #   A unique identifier for this payload. The UUID is encoded to include specific information.
+  #   See lib/msf/core/payload/uuid.rb in the https://github.com/rapid7/metasploit-framework repo.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] registered
```

Comment:
things like this relating to whether this has been used in a session should be deleted. a bool is not enough to really reflect how this was used in a session, so we'd want to use session logs to instead refer to this, instead of a bool here. Please remove (and remove from framework, since nothing there uses this either other than for setting).

Reply:
I suspect this was an artifact of this feature originally being designed for use without a database, so it needed to be able to mark info that the database should actually already be maintaining (whether you got a session). Can we add an association between the session logs and a payload?

Reply:
We can definitely add that association int he future. The logical model to associate with in the current architecture is `Mdm::SessionEvent`. That model could actually use a little more work to better track actions occurring within a session. We can complete both of these tasks at that point.